### DISCUSSION TITLE
CSSLints - Vocês usam?

### DISCUSSION THREAD

**Discussion by felipefialho** [frontendbr/forum#336] [Discussão]
Fala pessoal,

Estou querendo adicionar um CSSLint no meu workflow. Vocês usam?

Conhecem algum que integra bem com pré-processadores, em especial o Stylus?

**Comment by diessica** [comment]
Uso o [Stylelint](https://github.com/stylelint/stylelint).

**Comment by dsoldera** [comment]
Eu uso o scsslint (Sass Lint). Achei mais prático.

**Comment by thiamsantos** [comment]
Eu tenho usado [stylint](https://github.com/SimenB/stylint),  que é específico para stylus.

**Comment by felipefialho** [comment]
@thiamsantos Curti!

Vou testar.

**Comment by akfzambrana** [comment]
e sobre a configuração dos padrões? vcs seguem algum em especial?

**Comment by diessica** [comment]
O [stylelint-config-standard](https://github.com/stylelint/stylelint-config-standard) dá conta de muitas boas práticas com CSS que são encorajadas em styleguides como do Airbnb, Google, GitHub... 

Para convenções de organização de CSS, uso o [stylelint-rscss](https://github.com/rstacruz/stylelint-rscss) ou o [stylelint-selector-bem-pattern](https://github.com/davidtheclark/stylelint-selector-bem-pattern), dependendo da metodologia adotada no projeto.

**Comment by felipefialho** [comment]
Que foda @diessica! Valeu pelas dicas.

**Comment by thiamsantos** [comment]
Que legal @diessica, dá pra ver que o [stylelint](http://stylelint.io/) é muito bom!!!

Para o [stylint](https://github.com/SimenB/stylint) eu não achei nenhum padrão, por isso configurei ele da maneira que me é mais confortável :grin:.
- Meu [arquivo de configuração](https://github.com/thiamsantos/borg-boilerplate/blob/master/.stylintrc) do stylint
- [Breve descrição das regras](https://github.com/thiamsantos/borg-boilerplate#rules-enforced-on-stylus-code) que eu uso

**Comment by felipefialho** [comment]
@thiamsantos Configurei um padrão também.
Ja coloquei ele no Kratos ;)

Mas achei do carario esse pattern de BEM que a @diessica mostrou.

**Comment by bernardodiasc** [comment]
StyleLint extendido do standard aqui tambem, com algumas customizações.

**Comment by akfzambrana** [comment]
@diessica eu uso [este code styleguide](http://codeguide.co/) (mas arrumo as paradas manualmente 😔), pelo que olhei daria pra adaptar as configurações pra usar o Stylint ne?

**Comment by eduardojmatos** [comment]
Eu uso o [sass-lint](https://github.com/sasstools/sass-lint), mas o lance de trabalhar com um arquivo yml é meio ruim. Se fosse JSON seria mais bacana ainda :)

**Comment by bomffas** [comment]
@eduardojmatos sabe de uma solução que bloquei o processamento até que o erro seja corrigido ?

**Comment by eduardojmatos** [comment]
Então, o que o sass-lint faz é rodar a validação sobre os seus arquivos Sass e "estourar" os erros no seu terminal. Quando você diz processamento, do quê especificamente você fala? Processo de build? De teste?

**Comment by bomffas** [comment]
@eduardojmatos Digo quando o sass está sendo processado caso o lint encontre problema no arquivo ele só seja processado para css quando o problema for resolvido. Não sei se é possível fazer isso no deploy para homologação/produção.

**Comment by eduardojmatos** [comment]
@bfmjr saquei! Você pode rodar o sass-lint antes do compile do sass. Daí, no processo de build, se tiver algum erro de sass-lint ele nem executa o compile, saca?

Em homolog/prod dá pra seguir a mesma coisa

**Comment by bomffas** [comment]
@eduardojmatos Saquei, vou montar uma task no gulp e testar isso. Valeu 👍

**Comment by AlexsandroSA** [comment]
Que massa!!!
Estava procurando algo parecido com isso, agora vamos aos testes :)

**Comment by herberthenrique** [comment]
@bfmjr Você também pode usar o [Husky](https://github.com/typicode/husky) para rodar seu lint antes do `commit` ou `push` no git :)

**Comment by bomffas** [comment]
Valeu @herberthenrique vou testar
